main.go

```diff
@@ -209,6 +210,16 @@ func main() {
 			return
 		}
 
+		if cfg.auth.Authentication.Header.Enabled {
```

Comment:
@brancz I don't want `auth.Handle()` to have any knowledge about the outbound address, but as a result this there's Header handling in two different places. Any advice, which way is better? Thank you

Reply:
I think the way you did it is fine. We could potentially have auth return the auth information instead of silently attaching it to the request, but that would be outside of this PR anyways.